samza-core/src/main/java/org/apache/samza/startpoint/StartpointManager.java

```diff
@@ -0,0 +1,265 @@
+    startpointSerde.register(StartpointLatest.class);
+    startpointSerde.register(StartpointBootstrap.class);
+  }
+
+  /**
+   * Creates a {@link StartpointManager} instance with the provided {@link MetadataStoreFactory}
+   * @param metadataStoreFactory {@link MetadataStoreFactory} used to construct the underlying store.
+   * @param config {@link Config} required for the underlying store.
+   * @param metricsRegistry {@link MetricsRegistry} to hook into the underlying store.
+   * @return {@link StartpointManager} instance.
+   */
+  public static StartpointManager getWithMetadataStore(MetadataStoreFactory metadataStoreFactory, Config config, MetricsRegistry metricsRegistry) {
+    Preconditions.checkNotNull(metadataStoreFactory, "MetadataStoreFactory cannot be null");
+    Preconditions.checkNotNull(config, "Config cannot be null");
+    Preconditions.checkNotNull(metricsRegistry, "MetricsRegistry cannot be null");
+
+    MetadataStore metadataStore = metadataStoreFactory.getMetadataStore(NAMESPACE, config, metricsRegistry);
+
+    // TODO: Determine if it is worth it to make the duration expiration configurable
+    StartpointManager startpointManager = new StartpointManager(metadataStore, new StartpointSerde(), DEFAULT_EXPIRATION_DURATION);
+
+    LOG.info("StartpointManager created with metadata store: {}", metadataStore.getClass().getCanonicalName());
+
+    return startpointManager;
+  }
+
+  /**
+   * Starts the underlying {@link MetadataStore}
+   */
+  public void start() {
+    if (!started) {
+      metadataStore.init();
+      started = true;
+    } else {
+      LOG.warn("StartpointManager already started");
+    }
+  }
+
+  /**
+   * Writes a {@link Startpoint} that defines the start position for a {@link SystemStreamPartition}.
+   * @param ssp The {@link SystemStreamPartition} to map the {@link Startpoint} against.
+   * @param startpoint Reference to a Startpoint object.
+   */
+  public void writeStartpoint(SystemStreamPartition ssp, Startpoint startpoint) {
+    writeStartpointForTask(ssp, null, startpoint);
+  }
+
+  /**
+   * Writes a {@link Startpoint} that defines the start position for a {@link SystemStreamPartition} and {@link TaskName}.
+   * @param ssp The {@link SystemStreamPartition} to map the {@link Startpoint} against.
+   * @param taskName The {@link TaskName} to map the {@link Startpoint} against.
+   * @param startpoint Reference to a Startpoint object.
+   */
+  public void writeStartpointForTask(SystemStreamPartition ssp, TaskName taskName, Startpoint startpoint) {
+    Preconditions.checkState(started, "Underlying metadata store not available");
+    Preconditions.checkNotNull(ssp, "SystemStreamPartition cannot be null");
+    Preconditions.checkNotNull(startpoint, "Startpoint cannot be null");
+
+    try {
+      metadataStore.put(toStoreKey(ssp, taskName), startpointSerde.toBytes(startpoint));
+    } catch (Exception ex) {
+      throw new SamzaException(String.format(
```

Comment:
read/delete don't wrap the storage operation in a try-catch, so would it be ok to get rid of this one too?